src/shogun/multiclass/tree/CARTree.cpp

```diff
@@ -281,10 +281,30 @@ bool CARTree::train_machine(std::shared_ptr<Features> data)
 	{
 		prune_by_cross_validation(dense_features,m_folds);
 	}
-
 	return true;
 }
 
+void CARTree::compute_feature_importance(std::shared_ptr<bnode_t> node)
+{
+	auto left = node->left();
+	auto right = node->right();
```

Comment:
also could be `const auto& right = .. `?

Reply:
yes, i will change it.

Reply:
similarly here you avoid a copy, because this is just temporarily bound, e.g. the left and right variables are destroyed when you go out of scope. in theory you don't need to bind as a reference, as the compiler is required to do a copy elision here! So `const auto right = ` should suffice.
If you want to read more about references [this post is good](https://herbsutter.com/2020/02/23/references-simply/)

Reply:
ok! thanks for telling me this.